src/ResourceManager/SiteRecovery/Commands.SiteRecovery/Vault/SetAzureSiteRecoveryVaultSettings.cs

```diff
@@ -29,6 +29,13 @@ public class SetAzureSiteRecoveryVaultSettings : SiteRecoveryCmdletBase
         #region Parameters
 
         /// <summary>
+        /// Gets or sets vault Object.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.Default, Mandatory = true)]
+        [ValidateNotNullOrEmpty]
+        public System.Object Vault { get; set; }
```

Comment:
@markcowl  The problem is as follows. We've got two different resource types - "Site Recovery vault" and "Recovery services vault" . Look at Site Recovery vault as the ARM representation of our Classic vault and Recovery Services vault as our newer and better vault. The same set of PowerShell cmdlets work on both theses resources as essentially all exposed operations within them are the same. Our customer's don't really differentiate between the two resource types when it comes to the actions they perform in them, and to them both or either of them are just vaults. The particular cmdlet in question is used to set the session context of the recovery services vault or the site recovery vault within which all further operations are to be performed. Once this is done the actions are pretty much the same. 

We want to abstract out this distinction for our customers when they set the vault context.So we chose the vault parameter as a way to do this. With the objects for the two resource types being different we need a higher level abstratction or class from which we inherit to serve as the object type for this parameter - hence the choice of System Object.

As for how a customer creates this object, the object definition isnt something that they necessarily concern themselves with since this is never something they would have to create or manipulate by accessing the object. The vault is created using one of these cmdlets:-
- New-AzureRmSiteRecoveryVault or -New-AzureRmRecoveryServicesVault and can be retrieved later using the corresponding Get cmdlet. The returned object will be passed unmodified as an input to the cmdlet in question against the -Vault parameter and will get cast as a System Object

The ASR PowerShell user is typically an advanced customer who is familiar with the our UX and uses PowerShell to automate repeated tasks and is therefore unlikely to be confused at this step. Further, this will be more than adequately covered in our documentation